Objects/dictobject.c

```diff
@@ -6967,7 +7002,12 @@ _PyObject_TryGetInstanceAttribute(PyObject *obj, PyObject *name, PyObject **attr
 
     PyDictKeysObject *keys = CACHED_KEYS(Py_TYPE(obj));
     assert(keys != NULL);
-    Py_ssize_t ix = _PyDictKeys_StringLookup(keys, name);
+    Py_ssize_t ix;
+    if (keys->dk_kind == DICT_KEYS_SPLIT) {
+        ix = _PyDictKeys_StringLookupSplit(keys, name);
+    } else {
+        ix = _PyDictKeys_StringLookup(keys, name);
+    }
```

Comment:
Since `keys` is always the cached split keys object that is retrieved from the type object, I think this can just be:

```suggestion
    Py_ssize_t ix = _PyDictKeys_StringLookupSplit(keys, name);
```